libexec/probes/worker-scripts/uploader/rabbitmquploader.py

```diff
@@ -44,21 +44,28 @@ def publishSToMq(self, arguments, event_types, summaries, summaries_data):
     def SendMessagetoMQ(self, msg_body):
         # the max size limit in KB but python expects it in bytes                                                                           
         size_limit = self.maxMQmessageSize * 1000
-        size_msg = sys.getsizeof(json.dumps(msg_body))
+        size_msg = self.total_size(msg_body)
         # if size of the message is larger than 10MB discarrd                                                                             
         if size_msg > size_limit:
             self.add2log("Size of message body bigger than limit, discarding")
             return
-        # add to mq                                                                                                                       
-        try:
-            result = self.channel.basic_publish(exchange = self.exchange,
+        # add to mq
+        tries = 0
+        N = 5
+        while(tries<N):
+            tries = tries + 1
+            try:
+                result = self.channel.basic_publish(exchange = self.exchange,
                                                 routing_key = self.routing_key,
                                                 body = json.dumps(msg_body), 
                                                 properties = self.ch_prop)
-            if not result:
-                raise Exception('Exception publishing to rabbit MQ', 'Problem publishing to mq')
-        except Exception as e:
-            self.add2log("ERROR: Failed to send message to mq, exception was %s" % (e))
+                tries = N+1

```

Comment:
Replace this with `break`.

Try setting `result = None` before the loop.  This way, you only have to check once if the loop failed.